internal/appconfig/config_test.go

```diff
@@ -126,3 +127,32 @@ func TestConfigPortGetter(t *testing.T) {
 		})
 	}
 }
+
+// This can't go in helpers/clone_test.go because of an import cycle
+func TestCloneAppconfig(t *testing.T) {
+
+	config := &Config{
+		AppName: "testcfg",
+		RawDefinition: map[string]any{
+			"mounts": []Volume{
+				{
+					Source:      "src-raw",
+					Destination: "dst-raw",
+				},
+				{
+					Source:      "src2",
+					Destination: "dst2",
+				},
+			},
+		},
+		Mounts: &Volume{
+			Source:      "src",
+			Destination: "dst",
+		},
+		defaultGroupName: "some-group",
+	}
+
+	cloned := helpers.Clone(config)
+
+	assert.Equal(t, config, cloned)
```

Comment:
wdyt if the test modifies a few deep fields in the source before comparing, so we can be sure the cloned instance doesn't hold any references to the source

Reply:
good call!